lightning/src/blinded_path/message.rs

```diff
@@ -383,10 +389,17 @@ pub(super) fn blinded_hops<T: secp256k1::Signing + secp256k1::Verification>(
 			Some(scid) => NextMessageHop::ShortChannelId(scid),
 			None => NextMessageHop::NodeId(pubkey),
 		})
-		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { next_hop, next_blinding_override: None }))
-		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })));
+		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { padding: None, next_hop, next_blinding_override: None }))
+		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs { padding: None, context: Some(context) })));
 
-	let path = pks.zip(tlvs);
+	let max_length = tlvs.clone()
+		.map(|tlv| tlv.serialized_length())
+		.max()
+		.unwrap_or(0);
+
+	let length_tlvs = tlvs.map(|tlv| tlv.pad_to_length(max_length));
+
+	let path = pks.zip(length_tlvs);
```

Comment:
`length_tlvs` doesn't really describe what this is holding. Let's just combine this with the variable below to avoid naming altogether.

```rust
let path = pks.zip(tlvs.map(|tlv| tlv.pad_to_length(max_length)));
```